robotouille/robotouille_simulator.py

```diff
@@ -1,9 +1,97 @@
 import pygame
 from utils.robotouille_input import create_action_from_control
 from robotouille.robotouille_env import create_robotouille_env
+import asyncio
+import json
+import pickle
+import base64
+import websockets
+import time
+from pathlib import Path
+from datetime import datetime
+import imageio
+import traceback
+import networking.server as robotouille_server
+import networking.client as robotouille_client
 
+def simulator(environment_name: str, seed: int=42, role: str="simulator", display_server: bool=False, host: str="ws://localhost:8765", recording: str="", noisy_randomization: bool=False):
```

Comment:
biggest feedback is we want to move as much of the networking related code to the networking directory. I'd like to keep this file as clean as possible to the original so someone that looks at this function can easily reason how Robotouille works without needing to have a coupled understanding on the networking. I suggest moving these if statements to a function call within the `networking` directory and calling `simulate` immediately if the role is appropriate. Here is some psuedocode for what I have in mind to

```
def simulator(...):
"""...docstring..."""
if role != "local":
  # Run networked Robotouille
  networked_simulate(...)
simulate(...)
```
roles `replay` and `render` also make sense under networking since they're utilities for replaying or rendering data generated by the networking class.